File: Sources 4.5.2/Gizmox.WebGUI.Forms.CompanionKit/Controls/CheckBox/Functionality/ThreeStatePage.vb
Namespace CompanionKit.Controls.CheckBox.Functionality

	Public Class ThreeStatePage

		Public Sub New()

			' This call is required by the Visual WebGui UserControl Designer.
			InitializeComponent()
            'Populate ComboBox with the values of CheckState property
            mobjStateCombo.DataSource = [Enum].GetValues(GetType(CheckState))

            'Update Label showing current CheckState
            UpdateLabelText()

		End Sub


        ''' <summary>
        ''' Handles the SelectedIndexChanged event of the mobjStateCombo control.
        ''' </summary>
        ''' <param name="sender">The source of the event.</param>
        ''' <param name="e">The <see cref="System.EventArgs"/> instance containing the event data.</param>
        Private Sub mobjStateCombo_SelectedIndexChanged(sender As System.Object, e As System.EventArgs) Handles mobjStateCombo.SelectedIndexChanged

            'Change CheckBox.CheckState according to selected item in ComboBox
            mobjCheckBox.CheckState = DirectCast(mobjStateCombo.SelectedItem, CheckState)
        End Sub

        ''' <summary>
        ''' Updates the label text to show current CheckState property value.
        ''' </summary>
        Private Sub UpdateLabelText()
            mobjStateLabel.Text = "CheckBox State: " + mobjCheckBox.CheckState.ToString()
        End Sub

        ''' <summary>
        ''' Handles the CheckStateChanged event of the mobjCheckBox control.
        ''' </summary>
        ''' <param name="sender">The source of the event.</param>
        ''' <param name="e">The <see cref="System.EventArgs"/> instance containing the event data.</param>
        Private Sub mobjCheckBox_CheckStateChanged(sender As System.Object, e As System.EventArgs) Handles mobjCheckBox.CheckStateChanged
            'Change selected item in ComboBox according to current CheckBox.CheckState
            mobjStateCombo.SelectedItem = mobjCheckBox.CheckState

            'Update Label showing current CheckState
            UpdateLabelText()
        End Sub
    End Class

End Namespace

File: Sources 4.5.2/Gizmox.WebGUI.Forms.CompanionKit/Controls/CheckBox/Functionality/ThreeStatePage.cs
#region Using

using System;
using System.Collections.Generic;
using System.ComponentModel;
using System.Data;
using System.Drawing;
using System.Text;
using Gizmox.WebGUI.Common;
using Gizmox.WebGUI.Forms;

#endregion

namespace CompanionKit.Controls.CheckBox.Functionality
{
    public partial class ThreeStatePage : UserControl
    {
        public ThreeStatePage()
        {
            InitializeComponent();

            //Populate ComboBox with the values of CheckState property
            mobjStateCombo.DataSource = Enum.GetValues(typeof(CheckState));

            //Update Label showing current CheckState
            UpdateLabelText();
        }

        private void mobjStateCombo_SelectedIndexChanged(object sender, EventArgs e)
        {
            //Change CheckBox.CheckState according to selected item in ComboBox
            mobjCheckBox.CheckState = (CheckState)mobjStateCombo.SelectedItem;
        }

        /// <summary>
        /// Handles the CheckStateChanged event of the mobjCheckBox control.
        /// </summary>
        /// <param name="sender">The source of the event.</param>
        /// <param name="e">The <see cref="EventArgs"/> instance containing the event data.</param>
        private void mobjCheckBox_CheckStateChanged(object sender, EventArgs e)
        {
            //Change selected item in ComboBox according to current CheckBox.CheckState
            mobjStateCombo.SelectedItem = mobjCheckBox.CheckState;

            //Update Label showing current CheckState
            UpdateLabelText();
        }

        /// <summary>
        /// Updates the label text to show current CheckState property value.
        /// </summary>
        private void UpdateLabelText()
        {
            mobjStateLabel.Text = "CheckBox State: " + mobjCheckBox.CheckState.ToString();
        }
    }
}